### PR TITLE
[T2A4][T16-A1]Lai Zhiyuan

### PR DESCRIPTION
**Pull request by zyjames**
Ready for review

### CODE REVIEW

File: .settings/org.eclipse.jdt.ui.prefs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+eclipse.preferences.version=1
```
Comment (okkhoy) [inline .settings/org.eclipse.jdt.ui.prefs:1]: please use a proper gitignore file to eliminate these files from version control, if necessary

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -418,6 +428,40 @@ private static String getMessageForSuccessfulAddPerson(HashMap<PersonProperty, S
         return String.format(MESSAGE_ADDED,
                 getNameFromPerson(addedPerson), getPhoneFromPerson(addedPerson), getEmailFromPerson(addedPerson));
     }
+    
+    /**
+     * Adds a person (specified by the command args) to the address book.
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:433]: ouch! the header comment says it is add person, while the method is edit. 
please be careful when adding new methods/functionality to existing code. 
the header comments (javadocs documentation) should reflect the new change.

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -418,6 +428,40 @@ private static String getMessageForSuccessfulAddPerson(HashMap<PersonProperty, S
...
+    private static String executeEditPerson(String commandArgs) {
+        // try decoding a person from the raw args
+        final Optional<HashMap<PersonProperty, String>> decodeResult = decodePersonFromString(commandArgs);
+
+        // checks if args are valid (decode result will not be present if the person is invalid)
+        if (!decodeResult.isPresent()) {
+            return getMessageForInvalidCommandInput(COMMAND_EDIT_WORD, getUsageInfoForEditCommand());
+        }
+
+        // add the person as specified
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:448]: should be edit, no?

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -816,6 +860,38 @@ private static void addPersonToAddressBook(HashMap<PersonProperty, String> perso
...
     }
+    
+    /**
+     * Edits a person in the address book. Saves changes to storage file.
+     *
+     * @param updated person to edit
+     */
+    private static void editPersonToAddressBook(HashMap<PersonProperty, String> person) {
+    	if (updatePersonChanges(person)) {
+    		savePersonsToFile(getAllPersonsInAddressBook(), storageFilePath);
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:871]: indentation is off here.
we require the code to be indented with 4 spaces (not tabs)
Tip: You can get eclipse to do this for you; check out how?

File: src/seedu/addressbook/AddressBook.java [hunk 5]
```diff
@@ -544,6 +555,63 @@ private static String executeListAllPersonsInAddressBook() {
...
     }
+    
+    /**
+     * Displays all sorted persons in the address book to the user in asc/desc order.
+     *
+     * @return feedback display message for the operation result
+     */
+    private static String executeListAllPersonsInAddressBook(String commandArgs) {
+        if (!isSortPersonsArgValid(commandArgs)) {
+        	return getMessageForInvalidCommandInput(COMMAND_SORT_WORD, getUsageInfoForSortCommand());
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:610]: indentation is off here, again!

File: src/seedu/addressbook/AddressBook.java [hunk 6]
```diff
@@ -544,6 +555,63 @@ private static String executeListAllPersonsInAddressBook() {
...
+    /**
+     * Sorts all persons in the address book to the user.
+     *
+     * @return the sorted persons list in asc/desc order
+     */
+    private static ArrayList<HashMap<PersonProperty, String>> sortAllPersonsInAddressBookAsc(ArrayList<HashMap<PersonProperty, String>> unsortedPersonList,
+    		String arg) {
+    	if (arg.equals(COMMAND_SORT_PARAMETER_DESC)) {
+    		Collections.sort(unsortedPersonList, (HashMap<PersonProperty, String> person1, HashMap<PersonProperty, String> person2) 
+    				-> getNameFromPerson(person2).compareTo(getNameFromPerson(person1)));
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:652]: this way of splitting the line is not according to the coding standards we follow.
please take a look at the coding guidelines document and see how you can split a long line.

File: src/seedu/addressbook/AddressBook.java [hunk 7]
```diff
@@ -1086,16 +1155,22 @@ private static String getUsageInfoForViewCommand() {
 
     /** Returns string for showing 'help' command usage instruction */
     private static String getUsageInfoForHelpCommand() {
-        return String.format(MESSAGE_COMMAND_HELP, COMMAND_HELP_WORD, COMMAND_HELP_DESC)
-                + String.format(MESSAGE_COMMAND_HELP_EXAMPLE, COMMAND_HELP_EXAMPLE);
+        return String.format(MESSAGE_COMMAND_HELP, COMMAND_HELP_WORD, COMMAND_HELP_DESC) + LS
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java]: can think of using a more meaningful name for `LS`

File: README.md [hunk 8]
```diff
@@ -77,13 +77,19 @@ Examples:
 > Shows a list of persons, as an indexed list, in the order they were added to the address book, 
 oldest first.
 
-Format: `list`  
+Format: `list`
+
+#### Sorting all persons: `sort`
+
+> Shows a list of persons, as an indexed list, sorted in alphabetical order ascending/descending.
```
Comment (okkhoy) [inline README.md:84]: good to see you have updated the documentation to reflect the new feature :+1:

**Review by okkhoy — COMMENTED** [review]
it is good to see you have updated the documentation and help files on adding the new command (or feature) :+1: 
however, you also need to update the tests :-)

**Comment by okkhoy** [comment]
@zyjames some comments added.

Avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.

please close the PR once you have noted the comments.

File: src/seedu/addressbook/AddressBook.java (hunk 2 above)
Comment (zyjames) [reply src/seedu/addressbook/AddressBook.java:433]: Oops! My apologies for missing out on that!

File: src/seedu/addressbook/AddressBook.java (hunk 3 above)
Comment (zyjames) [reply src/seedu/addressbook/AddressBook.java:448]: Apologies for missing out on that too!

File: src/seedu/addressbook/AddressBook.java (hunk 5 above)
Comment (zyjames) [reply src/seedu/addressbook/AddressBook.java:610]: Thanks for pointing out the mistakes! Will be more careful in the future!

File: src/seedu/addressbook/AddressBook.java (hunk 4 above)
Comment (zyjames) [reply src/seedu/addressbook/AddressBook.java:871]: Apologies, I already configured my eclipse to do that already! Its my personal habit of pressing the tab I enter a new line. Will take note of it!

File: src/seedu/addressbook/AddressBook.java (hunk 6 above)
Comment (zyjames) [reply src/seedu/addressbook/AddressBook.java:652]: Ok! Thanks for pointing out the mistake!

**Pull request closed without merging**